packages/compiler-sfc/src/compileTemplate.ts

```diff
@@ -65,10 +65,19 @@ export function compileTemplate(
   const preprocessor =
     preprocessLang && consolidate[preprocessLang as keyof typeof consolidate]
   if (preprocessor) {
-    return doCompileTemplate({
-      ...options,
-      source: preprocess(options, preprocessor)
-    })
+    try {
+      return doCompileTemplate({
+        ...options,
+        source: preprocess(options, preprocessor)
+      })
+    } catch (e) {
+      return {
+        code: `export default function render() {}`,
+        source: options.source,
+        tips: [],
+        errors: [e]
+      }
+    }
```

Comment:
Added a try-catch for `preprocess`, not sure if it was intended to not be try-cached in [component-compiler-utils](https://github.com/vuejs/component-compiler-utils/blob/master/lib/compileTemplate.ts#L44-L48).

The error from a preprocessor is arbitrary - should we keep it as-is, coerce to string, or try to normalize it?

Reply:
It's probably fine to keep it as-is.